File: 04_strings_unicode/python/utf8.py
import sys
ascii = "Hello"
mem_size = sys.getsizeof(ascii)
print(f"'{ascii}': length: {len(ascii)} chars: {len(ascii)} mem_size: {mem_size}")

# make bytes
print(ascii.encode('utf-8'))
print("after e:", ascii[2:])

#
print("---")

uni = "Héllö"
mem_size = sys.getsizeof(uni)
print(f"'{uni}': length: {len(uni)} chars: {len(uni)} mem_size: {mem_size}")

# make bytes
#
#
print(uni.encode('utf-8'))
print("after e:", uni[2:])

File: 04_strings_unicode/rust/utf8.rs
fn main() {
    let ascii = String::from("Hello");
    println!(
        "'{}': length: {} chars: {} memsize: {}",
        ascii,
        ascii.len(),
        ascii.chars().count(),
        std::mem::size_of::<String>() + ascii.len()
    );
    println!("{:?}", ascii.as_bytes());
    println!("after e: {:?}", &ascii[2..]);

    println!("---");

    let uni = String::from("Héllö");
    println!(
        "'{}': length: {} chars: {} memsize: {}",
        uni,
        uni.len(),
        uni.chars().count(),
        std::mem::size_of::<String>() + uni.len()
    );
    println!("{:?}", uni.as_bytes());
    println!("after é: {:?}", &uni[2..]);
}
